homeassistant/components/sensor/geizhals.py

```diff
@@ -0,0 +1,150 @@
+"""
+Parse prices of a device from geizhals.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/sensor.geizhals/
+"""
+import logging
+from datetime import timedelta
+
+import voluptuous as vol
+
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+import homeassistant.helpers.config_validation as cv
+from homeassistant.util import Throttle
+from homeassistant.helpers.entity import Entity
+
+REQUIREMENTS = ['beautifulsoup4==4.6.0']
+_LOGGER = logging.getLogger(__name__)
+
+CONF_name = 'name'
+CONF_friendly_name = 'friendly_name'
+CONF_product_id = 'product_id'
+CONF_protocol = 'protocol'
+CONF_domain = 'domain'
+CONF_regex = 'regex'
+
+ICON = 'mdi:coin'
+MIN_TIME_BETWEEN_UPDATES = timedelta(seconds=120)
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_name): cv.string,
+    vol.Required(CONF_friendly_name, default='Price'): cv.string,
+    vol.Required(CONF_product_id): cv.positive_int,
+    vol.Optional(CONF_protocol, default='https'): vol.In(['https', 'http']),
+    vol.Optional(CONF_domain, default='geizhals.de'): vol.In(
+        ['geizhals.at',
+         'geizhals.eu',
+         'geizhals.de',
+         'skinflint.co.uk',
+         'cenowarka.pl']),
+    vol.Optional(CONF_regex, default='\D\s(\d*)[\,|\.](\d*)'): cv.string,
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Setup the Geizwatch Sensor."""
+    name = config.get(CONF_name)
+    friendly_name = config.get(CONF_friendly_name)
+    product_id = config.get(CONF_product_id)
+    protocol = config.get(CONF_protocol)
+    domain = config.get(CONF_domain)
+    regex = config.get(CONF_regex)
+
+    add_devices([Geizwatch(name, friendly_name, product_id, protocol, domain,
+                           regex)])
+
+
+class Geizwatch(Entity):
+    """Implementation of Geizwatch."""
+
+    def __init__(self, name, friendly_name, product_id, protocol, domain,
+                 regex):
+        """Initialize the sensor."""
+        self._name = name
+        self.friendly_name = friendly_name
+        self.data = GeizParser(product_id, protocol, domain, regex)
+        self.update()
```

Comment:
Use `add_devices(devices, True)` instead of `self.update()`.

-> https://home-assistant.io/developers/code_review_platform/#5-entity